quality_control_stock/tests/test_quality_control_stock.py

```diff
@@ -0,0 +1,222 @@
+# -*- coding: utf-8 -*-
+# (c) 2015 Oihane Crucelaegui - AvanzOSC
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from openerp.tests.common import TransactionCase
+
+
+class TestQualityControl(TransactionCase):
+
+    def setUp(self):
+        super(TestQualityControl, self).setUp()
+        self.picking_model = self.env['stock.picking']
+#         self.operation_model = self.env['stock.pack.operation']
+#         self.transfer_details_model = self.env['stock.transfer_details']
+        self.inspection_model = self.env['qc.inspection']
+        self.qc_trigger_model = self.env['qc.trigger']
+        self.picking_type_model = self.env['stock.picking.type']
+        self.product = self.env.ref('product.product_product_4')
+        self.partner1 = self.env.ref('base.res_partner_2')
+        self.partner2 = self.env.ref('base.res_partner_4')
+        self.test = self.env.ref('quality_control.qc_test_1')
+        self.picking_type = self.env.ref('stock.picking_type_out')
+        self.sequence = self.env.ref('stock.seq_type_picking_out')
+        inspection_lines = (
+            self.inspection_model._prepare_inspection_lines(self.test))
+        self.inspection1 = self.inspection_model.create({
+            'name': 'Test Inspection',
+            'inspection_lines': inspection_lines,
+        })
+        self.trigger = self.qc_trigger_model.search(
+            [('picking_type', '=', self.picking_type.id)])
+        self.lot = self.env['stock.production.lot'].create({
+            'name': 'Lot for tests',
+            'product_id': self.product.id,
+        })
+        move_vals = {
+            'name': self.product.name,
+            'product_id': self.product.id,
+            'product_uom': self.product.uom_id.id,
+            'location_id': self.picking_type.default_location_src_id.id,
+            'location_dest_id': self.picking_type.default_location_dest_id.id,
+        }
+        self.picking1 = self.picking_model.create({
+            'partner_id': self.partner1.id,
+            'picking_type_id': self.picking_type.id,
+            'move_lines': [(0, 0, move_vals)],
+        })
+        self.picking1.action_confirm()
+        self.picking1.force_assign()
+        self.picking1.do_prepare_partial()
+        for line in self.picking1.pack_operation_ids.filtered(
+                lambda r: r.product_id == self.product):
+            line.write({'lot_id': self.lot.id})
+
+    def test_inspection_create_for_product(self):
+        self.product.qc_triggers = [(
+            0, 0, {
+                'trigger': self.trigger.id,
+                'test': self.test.id,
+            }
+        )]
+        self.picking1.do_transfer()
+        self.assertEqual(self.picking1.created_inspections, 1,
```

Comment:
You should check in all that the created inspection is for the selected test on the trigger.